File: unity/Assets/Scripts/Multiplayer/Matchmaking/MatchmakingPasswordUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking
{
    /// <summary>
    /// Internal request data for updating a matchmaking lobby password in Unity.
    /// </summary>
    [System.Serializable]
    internal class MatchmakingPasswordUpdateRequest
    {
        /// <summary>
        /// The new password for the lobby.
        /// </summary>
        public string password;

        /// <summary>
        /// Initializes a new MatchmakingPasswordUpdateRequest.
        /// </summary>
        /// <param name="password">The new password for the lobby.</param>
        public MatchmakingPasswordUpdateRequest(string password)
        {
            this.password = password;
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/MatchmakingPasswordUpdateRequest.java
package com.michitai.multiplayer.matchmaking;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Request for updating matchmaking lobby password.
 */
public class MatchmakingPasswordUpdateRequest {
    @JsonProperty("password")
    private String password;

    public MatchmakingPasswordUpdateRequest(String password) {
        this.password = password;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}
